app/main/server/app/ui/window-util.js

```diff
@@ -24,19 +26,51 @@ function centerWindowOnSelectedScreen(window, openOnActiveDisplay) {
       selectedDisplay = display;
       break;
     }
+  } else if (openOnDisplay) {
+    // specified screen ID, match ID against display object
+    const displays = screen.getAllDisplays();
```

Comment:
This could be more succinctly written with an Array.find() such as:

```js
selectedDisplay = screen.getAllDisplays()
	.find((display) => display.id === openOnDisplay);
```

If `screen.getAllDisplays()` doesn't return an array, you could use `Array.from(..getAllDisplays())`

Reply:
Done